redfish-core/lib/account_service.hpp

```diff
@@ -1541,6 +1541,130 @@ inline void updateUserProperties(
+                messages::internalError(asyncResp->res);
+            }
+            return;
+        }
+        getAcfProperties(asyncResp, messageFDbus);
+    },
+        "xyz.openbmc_project.Certs.ACF."
+        "Manager",
+        "/xyz/openbmc_project/certs/ACF", "xyz.openbmc_project.Certs.ACF",
+        "InstallACF", decodedAcf);
+}
+
+inline void triggerUnauthenticatedACFUpload(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    std::optional<nlohmann::json> oem)
+{
+    std::optional<nlohmann::json> ibm;
+    if (!redfish::json_util::readJson(*oem, asyncResp->res, "IBM", ibm))
+    {
+        BMCWEB_LOG_ERROR("Illegal Property ");
+        messages::propertyMissing(asyncResp->res, "IBM");
+        return;
+    }
+
+    std::optional<nlohmann::json> acf;
+    if (ibm)
+    {
+        if (!redfish::json_util::readJson(*ibm, asyncResp->res, "ACF", acf))
+        {
+            BMCWEB_LOG_ERROR("Illegal Property ");
+            messages::propertyMissing(asyncResp->res, "ACF");
+            return;
+        }
+    }
+
+    if (acf)
+    {
+        std::vector<uint8_t> decodedAcf;
+        std::optional<std::string> acfFile{};
+        if (!redfish::json_util::readJson(*acf, asyncResp->res, "ACFFile",
+                                          acfFile))
+        {
+            BMCWEB_LOG_ERROR("Illegal Property ");
+            messages::propertyMissing(asyncResp->res, "ACFFile");
+            return;
+        }
+
+        std::string sDecodedAcf;
+        if (!crow::utility::base64Decode(*acfFile, sDecodedAcf))
+        {
+            BMCWEB_LOG_ERROR("base64 decode failure ");
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        try
+        {
+            std::copy(sDecodedAcf.begin(), sDecodedAcf.end(),
+                      std::back_inserter(decodedAcf));
+        }
+        catch (const std::exception& e)
+        {
+            BMCWEB_LOG_ERROR("Exception thrown: {}", e.what());
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        crow::connections::systemBus->async_method_call(
+            [asyncResp, decodedAcf](const boost::system::error_code ec,
+                                    const std::variant<bool>& retVal) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("Failed to read ACFWindowActive property");
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            const bool* isACFWindowActive = std::get_if<bool>(&retVal);
+
+            if (isACFWindowActive == nullptr)
+            {
+                BMCWEB_LOG_ERROR("nullptr for ACFWindowActive");
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            if (*isACFWindowActive)
+            {
+                uploadACF(asyncResp, decodedAcf);
+                return;
+            }
+
+            BMCWEB_LOG_ERROR("ACF window not set to "
```

Comment:
Should this an WARNING, rather than ERROR, as it is the unexpected condition to proceed, rather than something wrong in service?

https://github.com/openbmc/bmcweb/blame/e715d14b258a764f0e41559fe4a5e9f7d2efe314/DEVELOPING.md#L223-L229C58